python/snewpy/_model_downloader.py

```diff
@@ -117,54 +116,84 @@ def from_zenodo(zenodo_id:str, model:str, filename:str):
     """
     zenodo_url = f'https://zenodo.org/api/records/{zenodo_id}'
     record = requests.get(zenodo_url).json()
-    # Search for model file string in Zenodo request for this record.
+    # Search for file string in Zenodo request for this record.
     file = next((_file for _file in record['files'] if _file['key'] == filename), None)
 
     # If matched, return a tuple of URL and checksum.Otherwise raise an exception.
     if file is not None:
         return file['links']['self'], file['checksum'].split(':')[1]
     else:
         raise MissingFileError(filename)
-
-def get_model_data(model: str, filename: str, path: str = model_path) -> Path:
+        
+class ModelRegistry:
     """Access model data. Configuration for each model is in a YAML file
-    distributed with SNEWPY.
+        distributed with SNEWPY.
+    """
+    def __init__(self, config_file:Path = None, local_path: str = model_path):
+        """
+        Parameters
+        ----------
+        config_file: YAML configuration file. If None (default) use the 'model_files.yml' from SNEWPY resources
+        local_path: local installation path (defaults to astropy cache).
+        """
+        if config_file is None:
+            context = open_text('snewpy.models', 'model_files.yml')
+        else:
+            context = open(config_file)
+        with context as f:
+            self.config = yaml.safe_load(f)
+        self.models = self.config['models']
+        self.local_path = local_path
+
+    def get_file(self, config_path:str, filename:str)->Path:
+        """Get the requested data file from the models file repository
+
+        Parameters
+        ----------
+        config_path : dot-separated path of the model in the YAML configuration (e.g. "ccsn.Bollig_2016")
+        filename : Name of simulation datafile, or a relative path from the model sub-directory
+    
+        Returns
+        -------
+        Path of downloaded file.
+        """
+        tokens = config_path.split('.')
+        config = self.models
+        for t in tokens:
+            config = config[t]
+        #store the model name
+        model = tokens[-1]
+        modconf = config
```

Comment:
Unless I’m missing something, this renaming is unnecessary—and potentially confusing, since the code below uses a mixture of both variable names.

Reply:
You're right, it's not needed at all. Fixed.